### PR TITLE
PT-2156 Fix tests for lib, part2

### PR DESCRIPTION
**Pull request by svetasmirnova**
- [ ] The contributed code is licensed under GPL v2.0
- [ ] Contributor Licence Agreement (CLA) is signed
- [ ] util/update-modules has been ran
- [ ] Documention updated
- [ ] Test suite update

### CODE REVIEW

File: bin/pt-ioprofile [hunk 1]
```diff
@@ -203,9 +204,15 @@ parse_options() {
...
+       GLOBAL_CONFIG=1
+      _parse_config_files "/etc/percona-toolkit/percona-toolkit.conf"
+       GLOBAL_CONFIG=0
+       _parse_config_files "/etc/percona-toolkit/$TOOL.conf"
       if [ "${HOME:-}" ]; then
-         _parse_config_files "$HOME/.percona-toolkit.conf" "$HOME/.$TOOL.conf"
+         GLOBAL_CONFIG=1
+         _parse_config_files "$HOME/.percona-toolkit.conf"
+         GLOBAL_CONFIG=0
+         _parse_config_files "$HOME/.$TOOL.conf"
```
Comment (EvgeniyPatlan) [inline bin/pt-ioprofile:215]: what is the logic firstly set GLOBAL_CONFIG=1 and then GLOBAL_CONFIG=0?
maybe else statement is missed?

File: bin/pt-mysql-summary [hunk 2]
```diff
@@ -406,7 +413,11 @@ _parse_command_line() {
          else
             spec=$(grep "^short form:-$opt\$" "$PT_TMPDIR"/po/* | cut -d ':' -f 1)
             if [ -z "$spec"  ]; then
-               continue
+               if [ 1 -eq $GLOBAL_CONFIG ]; then
```
Comment (EvgeniyPatlan) [inline bin/pt-mysql-summary]: better $GLOBAL_CONFIG -eq 1

**Review by EvgeniyPatlan — CHANGES_REQUESTED** [review]

File: bin/pt-ioprofile (hunk 1 above)
Comment (svetasmirnova) [reply bin/pt-ioprofile:215]: This change is part of lib/bash/parse_options.sh GLOBAL_CONFIG variable is used to distinguish config file, applied globally to the toolkit and which can contain options, not available for particular tool, for the tool-specific config. Please check code at line 479: https://github.com/percona/percona-toolkit/blob/4f2401832e0ccb28fa66c122523e8aaf892af384/lib/bash/parse_options.sh#L478

File: bin/pt-mysql-summary (hunk 2 above)
Comment (svetasmirnova) [reply bin/pt-mysql-summary]: Updated

File: bin/pt-ioprofile (hunk 1 above)
Comment (svetasmirnova) [reply bin/pt-ioprofile:215]: It is possible to rewrite the function to take GLOBAL_CONFIG as a parameter, but this function is called in many places/tools that rely on the old behavior.

**Review by EvgeniyPatlan — APPROVED** [review]
LGTM :+1:

File: t/lib/TableSyncer.t [hunk 3]
```diff
@@ -1020,6 +1023,7 @@ like(
    qr/Complete test coverage/,
    '_d() works'
 );
+#$dbh->do('DROP DATABASE issue_96');
```
Comment (saikumar-vs) [inline t/lib/TableSyncer.t]: required or not?

File: t/lib/bash/collect.sh [hunk 4]
```diff
@@ -16,6 +16,10 @@ source "$LIB_DIR/safeguards.sh"
 source "$LIB_DIR/alt_cmds.sh"
 source "$LIB_DIR/collect.sh"
 
+# We need flush tables, otherwise we won't have stable results for opentables tests
+CMD_MYSQL="$(_which mysql)"
```
Comment (saikumar-vs) [inline t/lib/bash/collect.sh:20]: _which ?

File: t/lib/bash/collect.sh [hunk 5]
```diff
@@ -126,14 +121,48 @@ for file in $p-*; do
...
 
 is "$empty_files" "0" "No empty files"
 
+# ###########################################################################
+# Debug option for mysqladmin is not default now, we will test it separately.
+# ###########################################################################
+
+#Skipping until PT-2242 is fixed
+if false; then
+   parse_options "$BIN_DIR/pt-stalk" --run-time 2 -- --defaults-file=/tmp/12345/my.sandbox.cnf
```
Comment (saikumar-vs) [inline t/lib/bash/collect.sh:141]: im not sure of parse_options arguments but just "--" dosent seem right?, i it works pls ignore this comment :)

File: t/pt-mysql-summary/pt-mysql-summary_encryption.t [hunk 6]
```diff
@@ -131,6 +131,7 @@ $master_dbh->do("CREATE TABLE test.t2(a INT PRIMARY KEY) ENCRYPTION='Y'");
 
 $out = `bash $trunk/bin/$tool --list-encrypted-tables -- --defaults-file=/tmp/12345/my.sandbox.cnf`;
 
+#diag($out);
```
Comment (saikumar-vs) [inline t/pt-mysql-summary/pt-mysql-summary_encryption.t]: nitpick, pls remove if editing file :)

File: lib/bash/report_mysql_info.sh [hunk 7]
```diff
@@ -1292,30 +1272,31 @@ parse_wsrep_provider_options () {
...
- 
+   local instances_file="$1"
+   local variables_file="$2"
+   local GENERAL_JEMALLOC_STATUS=0
+   
+   for pid in $(grep '/mysqld ' "$instances_file" | awk '{print $1;}'); do
+	  local jemalloc_status="$(get_var "pt-summary-internal-jemalloc_enabled_for_pid_${pid}" "${variables_file}")"
+	  if [ -z $jemalloc_status ]; then
+		  continue
+      elif [ $jemalloc_status = 0 ]; then
```
Comment (saikumar-vs) [inline lib/bash/report_mysql_info.sh:1283]: nitpick indentation :)

File: t/lib/bash/collect.sh (hunk 4 above)
Comment (svetasmirnova) [reply t/lib/bash/collect.sh:20]: Yes, this is Tookit function, defined in lib/bash/alt_cmds.sh

File: t/lib/bash/collect.sh (hunk 5 above)
Comment (svetasmirnova) [reply t/lib/bash/collect.sh:141]: It works and passes arguments after -- to mysqld

File: t/lib/TableSyncer.t (hunk 3 above)
Comment (svetasmirnova) [inline t/lib/TableSyncer.t]:
```suggestion
```

File: t/pt-mysql-summary/pt-mysql-summary_encryption.t (hunk 6 above)
Comment (svetasmirnova) [inline t/pt-mysql-summary/pt-mysql-summary_encryption.t]:
```suggestion
```

**Review by saikumar-vs — APPROVED** [review]
Thanks , awsome huge patch :) and nice work.

**Pull request merged**